core/src/main/java/de/evoila/cf/broker/util/ServiceInstanceUtils.java

```diff
@@ -96,4 +98,28 @@ public static String portList(List<ServerAddress> serverAddresses) {
     public static ServerAddress serverAddress(String name, String host, int port) {
         return new ServerAddress(name, host, port);
     }
+
+    /**
+     * Checks whether an update with the given ServiceInstanceUpdateRequest would effectively change the service instance.
+     *
+     * This method heavily relies on the {@linkplain Object#equals(Object)} method to check for equality
+     * and resulting non-effective changes to a field. So it is mandatory for all objects contained in
+     * {@linkplain ServiceInstance#getParameters()} to have an overwritten equals method or can ensure equality by identity.
+     * @param serviceInstance the service instance object to compare with the update request
+     * @param request the update request object to compare with the service instance
+     * @return true if update would have effects and false if it would not
+     */
+    public static boolean isEffectivelyUpdating(ServiceInstance serviceInstance, ServiceInstanceUpdateRequest request) {
+        if (serviceInstance == null && request == null) return false;
+        if (serviceInstance == null ^ request == null) return true;
```

Comment:
If the service instance is `null` it shouldn't update. 
I hope the broker checks this beforehand and returns an error, but even so the return value should be `false` here.